### DISCUSSION TITLE
EQAVS1007E Error When attempting to run Debug from VS Code

### DISCUSSION THREAD

**Discussion by bigbearjeff63** [codefori/vscode-ibmi#2101] [IBM i] [answered]
i get the infamous "EQAVS1007E' error when attempting to run the Dubugger from VS Code. I suspect this is a certificate related issue. Is there any way I can get it to prompt to set these up? I must have missed it in the initial installation, and I can't seem to get back there to run it again. I've tried uninstalling and reinstalling and have had no luck. Any help would be greatly appreciated as I have a whole team here who'd love to be able to take advantage of this extension.

**Comment by sebjulliand** [comment]
Hi @bigbearjeff63 , could you share the entire message just to make sure we don't set you in the wrong direction? EQAVS1007E means the connection to the Debug Service failed, but it could be for a couple of reasons.

**Reply by bigbearjeff63** [reply]
Sure and thanks in advance for your help. 
EQAVS1007E 10.136.0.25 on port 8005 could not be connected. Message received: connect ECONNREFUSED 10.136.0.25:8005

**Reply by sebjulliand** [reply]
Right! So, this means that the Debug Service is not running or, if it's running, then it's not reachable over the network.
Let's check its state first. Open VSCode's `Run and Debug` view and then expand the `IBM i Debugger` view.
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/4caf746b-5dd8-45a5-be57-c810b02a9e20)
It will show the state of both the debug server and service.

What does it show for you?

If both are running, what tooltip do you get when you hover over the debug service?
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/32d0e4bb-fe17-4979-980c-efb5b223f8f8)

**Reply by bigbearjeff63** [reply]
Just to confirm. Are you wanting me to look under the "View" tab at the top and choose run and then Run and Debug? Or should I see something under the "Run" tab.

**Reply by bigbearjeff63** [reply]
Assuming I'm selecting Run and Debug under the View tab. A window pops up prompting with multiple options suggestion IBM i Debug. If I select that nothing happens.

**Reply by bigbearjeff63** [reply]
Just to be clear, I have not started any type of debugging service on the IBM i side.

**Reply by sebjulliand** [reply]
OK, so the Debug may not be running and not even configured. But that's what this view is here for.
Here is how you can open it, by hovering over your connection in the status bar at the bottom:
![NVIDIA_Share_esiFstnVAd-output](https://github.com/codefori/vscode-ibmi/assets/11096890/0c357b36-d8b5-4028-a182-5c07b0f75431)

Once this view is opened and loaded, it shows the status of both the Debug Server and the Debug Service. Both need to be running for the debug to work under VSCode. They can be started from here, and the Debug Service can be configured if needed from here as well (it will say so). You'll need an `*ALLOBJ` profile to start the server and service and to configure the service as well.

You can find all the information you need here in the documentation: https://codefori.github.io/docs/developing/debug/#configuring-the-debugger.

Let me know how it goes 😉

**Reply by bigbearjeff63** [reply]
Here is the result and it appears the server/service is not active. Can you walk me through getting that happening please?

![image](https://github.com/codefori/vscode-ibmi/assets/35222954/7cb74969-0112-4623-90ca-33da2d68123d)

**Reply by sebjulliand** [reply]
Click on the start button on each item to start them.
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/5a72a779-0d90-42d0-afef-02f8fbd50d17)

Once they have started (provided your profile allows you too), they should show like this:
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/95b6f4f1-4aab-446e-be7e-ac5e4f6d26b5)

**Reply by bigbearjeff63** [reply]
That seems like it got things going, but it appears when running debug that it's not respecting my library list. Do I have to start this from a CL that sets my Library list?

**Reply by sebjulliand** [reply]
No, if you debug from the object browser or IFS browser, then the library list and current library defined here should be picked up.
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/35d88097-5e55-4009-bfaf-c2aad3fd7f26)

The debug job is submitted in the `QSYSNOMAX` job queue, so you may be able to check what's going on in it (actual library list, job log, etc) once the debug has started and the first breakpoint is reached.

**Reply by bigbearjeff63** [reply]
okay thank you

**Reply by bigbearjeff63** [reply]
The error I'm getting is a CPF4203 and from what I can read on-line it indicates a debugger setting needs to be changed. It states "Either specify the INHWRT(*YES) parameter on the OVRDBF
command, or specify the UPDPROD(*YES) parameter on the CHGDBG command. Then
try your request again." Note: I am not using an OVRDBF command. How do I set the UPDPROD(*YES) value from Visual Studio?

**Reply by bigbearjeff63** [reply, marked as answer]
Ignore that! I found the setting. Thanks again for all of your help!
In a nutshell. I needed to start the Debug Service and the Debug Job from within Visual Studio by hovering over the system name I'm connected to and then clicking debugger. From there I hovered over Server and started it with the start icon to the right, and then I performed the same task for the Debug Service. After completing that, I was getting an error (CPF4203) as my application was updating production files. I once again hovered over my system name in the bottom task bar of the application and chose settings. From the settings screen there are Tabs across the top, and I chose the Debugger tab and then checked the Update Production Files check box. After that, everything worked great!

**Comment by sebjulliand** [comment]
You're very welcome 😄
